data-structures/linked_list/test_linked_list.py

```diff
@@ -52,3 +54,44 @@ def test_insert_after():
     ll = LinkedList(['a', 'b', 'c'])
     ll.insert_after('a', 'z')
     assert 'z' == ll.head._next.val
+
+
+def test_kkthFromEnd():
+    """
+    Checks if the linked-list is empty
+    Checks if out of range in Linked-List
+    Returns node with correct key value
+    """
+    nodes1 = LinkedList(['a', 1, 'hello world', 5.0, True])
+    assert nodes1.kthFromEnd(3) == 1
+
+    nodes2 = LinkedList([])
+    assert nodes2.kthFromEnd(5) == 'LinkedList is empty.'
+    nodes2.insert([5, 6, 7, 8, 9, 10])
+    assert nodes2.kthFromEnd(100) is False
+
+
+def test_ll_merge():
+    """
+    Validates if LinkedList is empty.
+    Returns Correct head node of LinkedList.
+    Returns the whole LinkedList merged.
+    """
+    arr1 = LinkedList()
+    arr2 = LinkedList()
+    LL = ll_merge(arr1, arr2)
+    assert LL is False
+
+    # Adding 6 to LinkedList
+    arr1 = LinkedList([6])
+    LL = ll_merge(arr1, arr2)
+    assert LL.head.val == 6
+
+    # Combining TWO LinkedList
+    arr1 = LinkedList([1, 3, 5, 7])
+    arr2 = LinkedList([2, 4, 6, 8])
+    LL = ll_merge(arr1, arr2)
+    assert LL.head.val.val == 1
```

Comment:
This does not make very much sense, as a node value should not be another node. Make sure you are not writing your tests to make your code pass, but rather your code to make your tests pass. Write what you expect to get back in your test rather than just matching what you are currently getting back to ensure that your code is being tested properly.